lib/src/main/java/global/eska/ddk/api/client/middleware/Middleware.java

```diff
@@ -0,0 +1,71 @@
+package global.eska.ddk.api.client.middleware;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import global.eska.ddk.api.client.model.ActionMessageCode;
+import global.eska.ddk.api.client.model.Headers;
+import global.eska.ddk.api.client.model.Message;
+import global.eska.ddk.api.client.model.MessageType;
+import global.eska.ddk.api.client.service.Blocker;
+import io.socket.client.Socket;
+import lombok.Getter;
+import lombok.Setter;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.UUID;
+
+@Getter
+@Setter
+@Service
+public class Middleware implements DDKMiddleware {
+
+    @Autowired
+    private ObjectMapper objectMapper;
+    @Autowired
+    private Blocker blocker;
+    private Message request;
+    private Message response;
+
+    public void send(Socket socket, ActionMessageCode code, JsonNode data) {
```

Comment:
Change JsonNode to Map. And if you need to serialize/deserialize object, you can do it here, not in service layer.